extensions/amp-embedly-card/0.1/amp-embedly-card-impl.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {getEmbedlyServiceForDoc} from './embedly-service';
+import {getIframe} from '../../../src/3p-frame';
+import {listenFor} from '../../../src/iframe-helper';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+/**
+ * Component tag identifier.
+ * @const {string}
+ */
+export const TAG = 'amp-embedly-card';
+
+/**
+ * Attribute name used to set api key.
+ * @const {string}
+ */
+export const API_KEY_ATTR_NAME = 'data-card-key';
+
+/**
+ * Implementation of the amp-embedly-card component.
+ * See {@link ../amp-embedly-card.md} for the spec.
+ */
+export class AmpEmbedlyCard extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    user().assert(
+        this.element.getAttribute('data-url'),
+        `The data-url attribute is required for <${TAG}> %s`,
+        this.element
+    );
+  }
+
+  /** @override */
+  layoutCallback() {
+    return getEmbedlyServiceForDoc(this.element).then(service => {
```

Comment:
I know I originally suggested a service, but there's be a better way.

1. Remove the service
2. Simply query for 
```js
document.querySelector('amp-embedly-key').getAttribute('data-key');
```

The service suggestion was way overdesigned, I apologize.

I also read the other thread regarding the API. Unless you have an objection and after this change, it looks good to me!

Reply:
Thanks Alan, I will be updating this PR tonight :D